service/src/main/java/uk/gov/hmcts/reform/fpl/service/document/ManageDocumentService.java

```diff
@@ -1157,4 +1163,83 @@ public PlacementEventData updatePlacementNoticesAdmin(CaseData caseData) {
+        Map<DocumentType, List<Element<NotifyDocumentUploaded>>> newDocumentsLA = new HashMap<>();
+        Map<DocumentType, List<Element<NotifyDocumentUploaded>>> newDocumentsCTSC = new HashMap<>();
+
+        Map<ConfidentialLevel, Map<DocumentType, List<Element<NotifyDocumentUploaded>>>> resultMapByConfidentialLevel =
+            Map.of(ConfidentialLevel.NON_CONFIDENTIAL, newDocuments,
+                ConfidentialLevel.LA, newDocumentsLA,
+                ConfidentialLevel.CTSC, newDocumentsCTSC);
+
+        for (DocumentType documentType : DocumentType.values()) {
+            for (ConfidentialLevel confidentialLevel : resultMapByConfidentialLevel.keySet()) {
+                if (documentType.getBaseFieldNameResolver() != null) {
+                    String fieldName = documentType.getBaseFieldNameResolver().apply(confidentialLevel);
+
+                    try {
+                        Map<DocumentType, List<Element<NotifyDocumentUploaded>>> newDocMap =
+                            resultMapByConfidentialLevel.get(confidentialLevel);
+
+                        List documentList = Optional.ofNullable(ObjectHelper
+                            .getFieldValue(caseData, fieldName, List.class)).orElse(List.of());
+                        List documentListBefore = Optional.ofNullable(ObjectHelper
+                            .getFieldValue(caseDataBefore, fieldName, List.class)).orElse(List.of());
+
+                        if (DocumentType.COURT_BUNDLE.equals(documentType)) {
+                            documentList = ((List<Element<HearingCourtBundle>>) documentList).stream()
+                                    .map(Element::getValue)
+                                    .map(HearingCourtBundle::getCourtBundle)
+                                    .flatMap(List::stream)
+                                    .collect(toList());
+
+                            documentListBefore = ((List<Element<HearingCourtBundle>>) documentListBefore).stream()
+                                    .map(Element::getValue)
+                                    .map(HearingCourtBundle::getCourtBundle)
+                                    .flatMap(List::stream)
+                                    .collect(toList());
+                        }
+
+                        for (Object document : documentList) {
+                            if (!documentListBefore.contains(document)) {
+                                List<Element<NotifyDocumentUploaded>> docList =
+                                    Optional.ofNullable(newDocMap.get(documentType))
+                                        .orElse(new ArrayList<>());
+                                newDocMap.putIfAbsent(documentType, docList);
+
+                                docList.add((Element<NotifyDocumentUploaded>) document);
+                            }
+                        }
+                    } catch (NoSuchMethodException e) {
```

Comment:
Think it is thrown by `ObjectHelper.getFieldValue`. Shall we just catch it for these two lines only?

                       ```
 List documentList = Optional.ofNullable(ObjectHelper
                            .getFieldValue(caseData, fieldName, List.class)).orElse(List.of());
                        List documentListBefore = Optional.ofNullable(ObjectHelper
                            .getFieldValue(caseDataBefore, fieldName, List.class)).orElse(List.of());
```